plugins/coverbrowser.koplugin/mosaicmenu.lua

```diff
@@ -731,6 +748,33 @@ function MosaicMenuItem:paintTo(bb, x, y)
         corner_mark:paintTo(bb, x+ix, y+iy)
     end
 
+    -- Paint Progressbar if needed
+    local bookinfo = BookInfoManager:getBookInfo(self.filepath, self.do_cover_image)
+    local percent_finished
+    local status
```

Comment:
We want all this in `MosaicMenuItem:update()` (same place as it is in `ListMenuItem:update()`).
(You don't want to query the sqlite db with getBookInfo() at each painting.)
Also, in :update(), just store the percentage as `self.percent_finished` like we store `self.been_opened` in `ListMenuItem:update()`, and just use this `self.percent_finished` in this `MosaicMenuItem:paintTo()`